starter_code/src/movies.js

```diff
@@ -2,14 +2,120 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(arr) {
+    let sortedMovies = [...arr];
+    sortedMovies.sort(function (a, b) {
+        if (a.year < b.year) {
+            return -1
+        } else if (a.year > b.year) {
+            return 1
+        } else {
+            if (a.title < b.title) {
+                return -1
+            } else if (a.title > b.title) {
+                return 1
+            } else {
+                return 0
+            }
+        }
+    });
+    return sortedMovies
+};
+
+//console.log(orderByYear(movies));
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movieArray, director, type) {
+    const filterResults = movieArray.filter(function(movie) {
+        return movie.director === director && movie.genre.indexOf(type) >= 0 // optional: .includes
+    });
+    //console.log(filterResults);
+    return filterResults.length
+}
+
+//console.log(howManyMovies(movies, "Steven Spielberg", "Drama"));
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movieArray) {
+    let sortedMovies = [...movieArray];
+    sortedMovies.sort(function (a, b) {
+            if (a.title > b.title) {
+                return -1
+            } else if (a.title < b.title) {
+                return 1
+            } else {
+                return 0
+            }
+        }
```

Comment:
Here you sort the array inversely alphabetically, you should invert greater and lower and it works